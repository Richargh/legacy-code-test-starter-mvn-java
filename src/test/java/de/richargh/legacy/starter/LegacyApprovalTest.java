package de.richargh.legacy.starter;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.PrintStream;

class LegacyApprovalTest {

    @Test
    void testWithApprovalTests() {
        CombinationApprovals.verifyAllCombinations(
                this::runWithResult,
                new String[]{"1"},
                new Boolean[]{true, false});
    }

    private String runWithResult(String tmp, boolean flag){
        try(var byteStream = new ByteArrayOutputStream(); var printStream = new PrintStream(byteStream)){
            System.setOut(printStream);
            run(tmp, flag);
            return byteStream.toString();
        } catch (IOException exception){
            throw new RuntimeException(exception);
        }
    }

    private static void run(String tmp, boolean flag) {
        Legacy.rand.setSeed(123);
        var testee = new Legacy();
        testee.doSth(tmp, flag);
    }
}
