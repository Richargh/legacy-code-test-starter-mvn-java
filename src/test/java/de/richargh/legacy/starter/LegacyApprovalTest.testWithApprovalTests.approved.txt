[1, true] => Setting mode 1
I am now in mode 1
You gain
and
 
[1, false] => and
 
